woocommerce/woocommerce
# [Settings] Render custom and slotfill settings types

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/52973

Render custom types using `dangerouslySetInnerHTML` and slotFills to be picked up by included JS scripts.

<img width="1122" alt="image" src="https://github.com/user-attachments/assets/585d88d5-5b75-4cc6-bc18-f7b5b144b69c" />



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add/Enable Gutenberg and turn on the `settings` feature flag
2. WooCommerce > Settings > Emails and see a custom HTML table rendered.
3. Next go to Site visibility to confirm our slotFill root is available by querying the document in the console.
```js
document.getElementById('wc_settings_site_visibility_slotfill');
```



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Render new Settings custom and slotfill type settings elements.

</details>
